flutter_frontend_server/BUILD.gn

```diff
@@ -42,62 +46,60 @@ if (is_fuchsia_host || is_fuchsia) {
       ":flutter_frontend_server",
     ]
   }
-} else {
-  import("//third_party/dart/utils/application_snapshot.gni")
-
-  frontend_server_files =
-      exec_script("//third_party/dart/tools/list_dart_files.py",
-                  [
-                    "absolute",
-                    rebase_path("."),
-                  ],
-                  "list lines")
-
-  frontend_server_files +=
-      exec_script("//third_party/dart/tools/list_dart_files.py",
-                  [
-                    "absolute",
-                    rebase_path("../../third_party/dart/pkg"),
-                  ],
-                  "list lines")
-
-  application_snapshot("frontend_server") {
-    main_dart = "bin/starter.dart"
-    deps = [
-      ":package_incremental_compiler",
-      "$flutter_root/lib/snapshot:kernel_platform_files",
-    ]
-    dot_packages = rebase_path(".packages")
-    flutter_patched_sdk = rebase_path("$root_out_dir/flutter_patched_sdk")
-    training_args = [
-      "--train",
-      "--sdk-root=$flutter_patched_sdk",
-      rebase_path(main_dart),
-    ]
+}
 
-    inputs = frontend_server_files
-  }
+frontend_server_files =
+    exec_script("//third_party/dart/tools/list_dart_files.py",
+                [
+                  "absolute",
+                  rebase_path("."),
+                ],
+                "list lines")
 
-  # For flutter/flutter#36738 we make the source files available so that
-  # we can generate a local frontend_server snapshot in the tools cache.
-  action("package_incremental_compiler") {
-    script = "$flutter_root/flutter_frontend_server/package_incremental.py"
+frontend_server_files +=
+    exec_script("//third_party/dart/tools/list_dart_files.py",
+                [
+                  "absolute",
+                  rebase_path("../../third_party/dart/pkg"),
+                ],
+                "list lines")
 
-    inputs = frontend_server_files
+application_snapshot("frontend_server") {
```

Comment:
Aren't you going to end up with two frontend_server app snapshots defined in fuchsia setting - one defined here, second defined at line 37?

Reply:
This target will not be invoked in fuchsia tree. So we should end up with only one. For context: none of the targets other that the one above are invoked in Fuchsia tree currently.

Reply:
What was the motivation to move these out of of `else` branch guarded by `is_fuchsia` then?

Reply:
There are a few things going on here:
1. When we build for Fuchsia target from the flutter tree, `is_fuchsia` is set to true. Which means that this needs to be outside the else block.
2. Though the target above is in `is_fuchsia` block, it is not invoked because the root of all the targets for Fuchsia builds in the engine tree is https://github.com/flutter/engine/blob/master/shell/platform/fuchsia/BUILD.gn#L68

Reply:
Okay, talked offline. It's a temporary thing until fuchsia stops building flutter_frontend_server itself,  instead uses app snapshot that is built by flutter engine and gets packaged for fuchsia consumption.